WcaOnRails/app/models/team_member.rb

```diff
@@ -13,7 +15,25 @@ def current_member?
   validate :start_date_must_be_earlier_than_end_date
   def start_date_must_be_earlier_than_end_date
     if start_date && end_date && start_date >= end_date
-      errors.add(:start_date, "must be earlier than end_date")
+      errors.add(:start_date, "must be earlier than end date")
+    end
+  end
+
+  validate :membership_periods_cannot_overlap_for_single_user
+  def membership_periods_cannot_overlap_for_single_user
+    if (start_date)
+      new_period = start_date..(end_date || Date::Infinity.new)
+      if self.id
+       team_members = TeamMember.where("user_id = :user_id and team_id = :team_id and id <> :team_member_id", {user_id: self.user_id, team_id: self.team_id, team_member_id: self.id})
+      else
+       team_members = TeamMember.where("user_id = :user_id and team_id = :team_id", {user_id: self.user_id, team_id: self.team_id})
+      end
+      team_members.each do |team_member|
+        existing_period = team_member.start_date..(team_member.end_date || Date::Infinity.new)
+        if new_period.overlaps? existing_period
+          errors.add(:user_id, "must not have overlapping membership dates with the #{self.team.name}.")
+        end
+      end
     end
```

Comment:
Hi @jonatanklosko,

I'm not sure I follow here. This validation is not for all members, but for a single member.

The new record which isn't saved yet has it's dates put into the `new_period` variable. Then I retrieve all other records for this `user_id` except the record this is updating (if it's an update where self.id exists). So the only new record is the one stored in self?

Let me know if that makes sense or if I am not quite understanding the process here.